### PR TITLE
feat: rewrite `not in` expr to `in`

### PR DESCRIPTION
**Pull request by jiacai2050**
## Rationale

Current filter is based on bloom-filter like structure, it can give accurate answer if an item doesn't exist in one collection, so by convert `col != value` to `col == value2`, we can fully utilize this feature.

## Detailed Changes
- Save column value to SST metadata when distinct value is less than 20.
- Rewrite not-like expr to its opposite.

## Test Plan
New UT.

### CODE REVIEW

File: analytic_engine/src/sst/parquet/writer.rs [hunk 1]
```diff
@@ -203,6 +225,38 @@ impl RecordBatchGroupWriter {
...
+            }
+
+            let col_values = match col_values {
+                None => continue,
+                Some(v) => v,
+            };
+            let rows_num = record_batch.num_rows();
+            let column_block = record_batch.column(col_idx);
+            for row_idx in 0..rows_num {
+                match col_values {
```
Comment (chunshao90) [inline analytic_engine/src/sst/parquet/writer.rs:248]: Check KEEP_COLUMN_VALUE_THRESHOLD in this loop.

File: analytic_engine/src/sst/parquet/row_group_pruner.rs [hunk 2]
```diff
@@ -212,4 +354,54 @@ mod tests {
...
+            (
+                // ip != 127.0.0.1
+                col("ip").not_eq(lit("127.0.0.1")),
+                col("ip").not_eq(lit("127.0.0.1")),
+            ),
+            (
+                // ip = 127.0.0.1
+                col("ip").eq(lit("127.0.0.1")),
+                col("ip").eq(lit("127.0.0.1")),
+            ),
```
Comment (chunshao90) [inline analytic_engine/src/sst/parquet/row_group_pruner.rs:401]: Add a test, col is `ip2`.

Comment (jiacai2050) [reply analytic_engine/src/sst/parquet/row_group_pruner.rs:401]: fixed

**Review by chunshao90 — APPROVED** [review]
LGTM

**Pull request merged**